Source/SwiftLintFramework/Rules/ColonRule.swift

```diff
@@ -94,42 +94,90 @@ public struct ColonRule: CorrectableRule, ConfigurationProviderRule {
             "func abc(↓def:  Void) {}\n": "func abc(def: Void) {}\n",
             "func abc(↓def :Void) {}\n": "func abc(def: Void) {}\n",
             "func abc(↓def : Void) {}\n": "func abc(def: Void) {}\n",
-            "func abc(def: Void, ↓ghi :Void) {}\n": "func abc(def: Void, ghi: Void) {}\n"
+            "func abc(def: Void, ↓ghi :Void) {}\n": "func abc(def: Void, ghi: Void) {}\n",
+            "let abc = [Void↓:Void]()\n": "let abc = [Void: Void]()\n",
+            "let abc = [Void↓ : Void]()\n": "let abc = [Void: Void]()\n",
+            "let abc = [Void↓:  Void]()\n": "let abc = [Void: Void]()\n",
+            "let abc = [Void↓ :  Void]()\n": "let abc = [Void: Void]()\n",
+            "let abc = [1: [3↓ : 2], 3: 4]\n": "let abc = [1: [3: 2], 3: 4]\n",
+            "let abc = [1: [3↓ : 2], 3↓:  4]\n": "let abc = [1: [3: 2], 3: 4]\n"
         ]
     )
 
     public func validateFile(_ file: File) -> [StyleViolation] {
-        return violationRangesInFile(file, withPattern: pattern).flatMap { range in
+        let violations = violationRangesInFile(file, withPattern: pattern).flatMap { range in
```

Comment:
The `violationRangesInFile(_:withPattern:)` function name should make it clearer that it only detects type violations.